src/uu/dd/src/dd.rs

```diff
@@ -824,15 +828,15 @@ impl<'a> Output<'a> {
     /// the output file is no longer needed. If not possible, then
     /// this function prints an error message to stderr and sets the
     /// exit status code to 1.
-    #[allow(unused_variables)]
+    #[cfg_attr(not(target_os = "linux"), allow(clippy::unused_self, unused_variables))]
     fn discard_cache(&self, offset: libc::off_t, len: libc::off_t) {
         #[cfg(target_os = "linux")]
         {
             show_if_err!(self.dst.discard_cache(offset, len).map_err_context(|| {
                 "failed to discard cache for: 'standard output'".to_string()
             }));
         }
-        #[cfg(target_os = "linux")]
+        #[cfg(not(target_os = "linux"))]
```

Comment:
it changed the behavior, no ?

Reply:
@sylvestre correct - I am pretty certain this was a bug in the code. Note that the code block is actually a noop, so it didn't show up in any testing, but this change makes sure that when code is added, it will be correctly compiled